### ISSUE TITLE
[BUG]

### ISSUE THREAD

**Issue by Snehaveeranna** [microsoft/playwright#22752]
### System info
- Playwright Version: [v1.XX]
- Operating System: [All, Windows 11, Ubuntu 20, macOS 13.2, etc.]
- Browser: [All, Chromium, Firefox, WebKit]
- Other info:

### Source code

- [ ] I provided exact source code that allows reproducing the issue locally.

**Link to the GitHub repository with the repro**

[https://github.com/your_profile/playwright_issue_title]

or

**Config file**

```js
// playwright.config.ts
import { defineConfig, devices } from '@playwright/test';

export default defineConfig({
  projects: [
    {
      name: 'chromium',
      use: { ...devices['Desktop Chrome'], },
    },
});
```

**Test file (self-contained)**

```js
it('should check the box using setChecked', async ({ page }) => {
  await page.setContent(`<input id='checkbox' type='checkbox'></input>`);
  await page.getByRole('checkbox').check();
  await expect(page.getByRole('checkbox')).toBeChecked();
});
```

**Steps**
- [Run the test]
- [...]

**Expected**

[Describe expected behavior]

**Actual**

[Describe actual behavior]

**Comment by Snehaveeranna** [comment]
I am using Playwright with Java with Cucumber framwork, i am unable to run my test sceanrios in parallel, though the browser opens parallelly it is not performing any actions in browsers.

**Comment by Snehaveeranna** [comment]
Also Playwright instance i mean browser is not closing after element is not found and due to this extent cucumber report is overlapping.

**Comment by yury-s** [comment]
Note that Playwright Java is not thread-safe, make sure you follow [this guide](https://playwright.dev/java/docs/multithreading) when writing parallel tests. Since you are using a Cucumber framework the execution is likely controlled by it and the questions should be addressed to the framework's forum. If you still think this is a bug in Playwright, file a new issue and fill out all the fields in the bug template including

* The version of Playwright Java you are using.
* A sample of your code, including the configuration for running tests in parallel.
* The specific error messages or logs, if any, that you're encountering.
